sqlglot/dialects/hive.py

```diff
@@ -485,7 +486,7 @@ class Generator(generator.Generator):
                 [
                     transforms.eliminate_qualify,
                     transforms.eliminate_distinct_on,
-                    transforms.unnest_to_explode,
+                    partial(transforms.unnest_to_explode, unnest_using_arrays_zip=False),
```

Comment:
See comment about using `self.unsupported` if we can't correctly transpile to Hive the case where `len(unnest.expressions) > 1`.

Reply:
`self.unsupported` is not accessible in `unnest_to_explode`. Is it possible to throw an exception in unnest_to_explode according to `unnest_using_arrays_zip`?

Reply:
We should aim to use `unsupported` instead of preemptively raising, because with the former you can control whether an exception will be thrown by setting the error level accordingly.

I'm not sure what's the best way to do this, but perhaps you could try overriding `select_sql` in Hive and invoking `unnest_to_explode` in it, unless you found there's at least an `Unnest` expression in scope with multiple `expressions`, in which case you'd call `self.unsupported(...)`. The `find_all_in_scope` helper under `optimizer/scope.py` may be useful here to avoid traversing subqueries and only look for unnests in the current `SELECT`'s scope. Note that this approach will affect subclasses of `Hive` (e.g. `Spark`) because they'll now use this `select_sql` override, so we need to avoid that somehow if that's the approach.

----

Before trying all of the above, as Vaggelis mentioned we should make sure Hive doesn't support `INLINE` or `ARRAYS_ZIP`.